src/logger/MetricsContext.ts

```diff
@@ -180,9 +180,9 @@ export class MetricsContext {
   public putMetric(key: string, value: number, unit?: Unit | string): void {
     const currentMetric = this.metrics.get(key);
     if (currentMetric) {
-      currentMetric.addValue(value);
+      currentMetric.addValue(Number(value));
```

Comment:
What do you think of throwing here if we find that `value` is not a number?
Javascript has weird/grey areas when it comes to parsing. It might make more sense to delegate the parsing to whoever is calling `putMetric`.

Reply:
Anything that isn’t just passing the value, quoting it and having cloudwatch silently drop it sounds fine to be honest.

I just didn’t want to make an API that currently fails silently start throwing exceptions, as that’s not a backwards compatible change (and so implies a new major version if semver is being used).

But if throwing an exception is the way that the AWS team wants to go then that’s fine with me!